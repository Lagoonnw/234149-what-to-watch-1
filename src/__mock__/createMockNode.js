export function createNodeMock(element) {
  if (element.type === `video`) {
    return {};
  }
  return null;
}
