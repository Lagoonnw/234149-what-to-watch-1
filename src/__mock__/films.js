export const films = [
  {
    id: 54585444,
    name: `Shutter Island`,
    genre: `thriller`,
    poster: `img/shutter-island.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 4587653,
    name: `Pulp Fiction`,
    genre: `crime`,
    poster: `img/pulp-fiction.jpg`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 54546847,
    name: `Snatch`,
    genre: `crime`,
    poster: `img/snatch.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 54545454545,
    name: `Bohemian Rhapsody`,
    genre: `drama`,
    poster: `img/bohemian-rhapsody.jpg`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 549963331,
    name: `Aviator`,
    genre: `family`,
    poster: `img/aviator.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

