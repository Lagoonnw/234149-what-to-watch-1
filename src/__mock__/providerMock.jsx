import React from 'react';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {genresList} from '../constants/constants';
import PropTypes from 'prop-types';

const mock = [
  {
    id: 54545454,
    name: `Snatch`,
    genre: `horror`,
    poster: `pic.jpg`,
    src: `video.mp4`,
    onMouseLeave: jest.fn(),
    onMouseEnter: jest.fn()
  },
  {
    id: 9898988,
    name: `Pulp Fiction`,
    genre: `horror`,
    poster: `pic.jpg`,
    src: `video.mp4`,
    onMouseLeave: jest.fn(),
    onMouseEnter: jest.fn()
  },
  {
    id: 545468787,
    name: `The Witcher`,
    genre: `horror`,
    poster: `pic.jpg`,
    src: `video.mp4`,
    onMouseLeave: jest.fn(),
    onMouseEnter: jest.fn()
  }
];
const initialStateMock = {
  activeGenre: null,
  movies: mock,
  sortedMovies: mock,
  genres: Array.from(genresList)
};
const reducer = (state = initialStateMock) => state;
const store = createStore(reducer);

export const ProviderMock = ({children}) => <Provider store={store}>{children}</Provider>;

ProviderMock.propTypes = {
  children: PropTypes.object
};
