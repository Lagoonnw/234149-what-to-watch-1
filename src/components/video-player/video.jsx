import React from 'react';
import PropTypes from 'prop-types';
import {DefaultVideoSize} from '../../constants/constants';

export const Video = (props) => {
  const {
    poster,
    src,
    muted,
    autoPlay,
    width,
    height,
    reference
  } = props;

  return (
    <video
      poster={poster}
      width={`${width}px`}
      height={`${height}px`}
      src={src}
      muted={muted}
      autoPlay={autoPlay}
      ref={reference}
      style={{objectFit: `cover`}}
    />
  );
};

Video.propTypes = {
  poster: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  muted: PropTypes.bool,
  autoPlay: PropTypes.bool,
  width: PropTypes.number,
  height: PropTypes.number,
  isPlaying: PropTypes.bool,
  controls: PropTypes.bool,
  reference: PropTypes.object.isRequired
};

Video.defaultProps = {
  muted: false,
  autoPlay: false,
  width: DefaultVideoSize.WIDTH,
  height: DefaultVideoSize.HEIGHT,
  controls: false
};

