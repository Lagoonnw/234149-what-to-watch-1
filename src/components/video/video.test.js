import React from 'react';
import renderer from 'react-test-renderer';
import {Video} from './video.jsx';
import {createNodeMock} from '../../__mock__/createMockNode';

describe(`Video player snapshot`, () => {
  test(`Should render video player correctly`, () => {
    const options = {createNodeMock};
    const ref = React.createRef();
    const tree = renderer.create(
        <Video
          poster={`pic.jpg`}
          src={`video.mp4`}
          reference={ref}/>, options).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
