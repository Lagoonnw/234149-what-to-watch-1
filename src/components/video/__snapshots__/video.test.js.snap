// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video player snapshot Should render video player correctly 1`] = `
<video
  autoPlay={false}
  controls={false}
  height="175px"
  muted={false}
  poster="pic.jpg"
  src="video.mp4"
  style={
    Object {
      "objectFit": "cover",
    }
  }
  width="280px"
/>
`;
